### ISSUE TITLE
Scan-manga FR - page not available

### ISSUE THREAD

**Issue by pixelroll** [keiyoushi/tachiyomi-extensions-source#2232]
### Source information

fr.scanmanga 1.4.7

### Source language

French

### Steps to reproduce

tried to open the source in tachiyomi and web browser (work in external browser but not in internal)

### Expected behavior

manga display on tachiyomi

### Actual behavior

no results found

### Mihon/Tachiyomi version

tachiyomiSY 1.10.5

### Android version

Android 13

### Acknowledgements

- [X] I have searched the existing issues and this is a new ticket, **NOT** a duplicate or related to another open or closed issue.
- [X] I have written a short but informative title.
- [X] I have updated the app to version **[0.15.3](https://github.com/mihonapp/mihon/releases/latest)**.
- [X] I have updated all installed extensions.
- [X] I have tried the [troubleshooting guide](https://tachiyomi.org/help/guides/troubleshooting/).
- [ ] I have admitted that I am a clown by having checked this box, as I have not read these acknowledgements.
- [X] If this is an issue with the app itself, I should be opening an issue in the [app repository](https://github.com/tachiyomiorg/tachiyomi/issues/new/choose).
- [X] I will fill out all of the requested information in this form.

**Comment by barrelltitor** [comment]
Website seems to be dead https://scan-manga.com/ https://www.scan-manga.com/

**Comment by pixelroll** [comment]
> Website seems to be dead https://scan-manga.com/ https://www.scan-manga.com/

open the link in a classic browser and not in tachiyomi and you will see that it works well!

[Screenshot_2024-04-03-01-07-12-665_com.android.chrome.jpg](https://github.com/keiyoushi/extensions-source/assets/32129359/9a68f2e8-f2f6-4e2a-9909-09237bc7d69a)

**Comment by barrelltitor** [comment]
![firefox_1KR8pX1fuG](https://github.com/keiyoushi/extensions-source/assets/108460132/467b13ae-2939-461a-b406-76edc5b39ffc)
404 for me on mobile and android, tried both chrome and firefox

Do they region restrict to france maybe?

**Comment by Smol-Ame** [comment]
Yeah, I'm just getting 404'd with both "scan-manga" and your "m.scan-manga" screenshot you had earlier, with my PC or phone (United States, but did try on phone with VPN to Netherlands & Poland)

Could you give a working link to a title on the source's website?

**Comment by KirinRaikage** [comment]
Nah, it works fine, maybe a region-blocking stuff. https://www.scan-manga.com/14160/Can-You-Hear-Me.html

**Comment by pixelroll** [comment]
I am in France and the site works very well on Firefox under Android 13 but on the browser integrated into tachiyomi it indicates the code: net::ERR_HTTP_RESPONSE_CODE_FAILURE

with a vpn in the united states : error 404

**Comment by UjCbFwtBayFM** [comment]
Same here. Website works fine using external browser but not in app or using WebView.

**Comment by pixelroll** [comment]
it's not geo blocked! @vetleledaal

**Comment by vetleledaal** [comment]
@pixelroll 
If it's not geo-blocking, then the site is down. Tested with desktop/mobile + Firefox/Chrome.

**Comment by pixelroll** [comment]
> @pixelroll 
> If it's not geo-blocking, then the site is down. Tested with desktop/mobile + Firefox/Chrome.

works on Firefox but not tachiyomi

![Screenshot_2024-08-20-11-35-19-374_org.mozilla.firefox.jpg](https://github.com/user-attachments/assets/303d84f0-20f2-4b24-b9a6-a18df96ff131)

**Comment by Mitia99** [comment]
@KirinRaikage Any update on this ?

**Comment by anjaragermai** [comment]
Hi, I'm from Madagascar and the site works well in an external browser but not on the app, the WebView doesn't do anything https://m.scan-manga.com/?po
![Screenshot_20241217-104733](https://github.com/user-attachments/assets/fe8f6b87-df52-4c88-936d-179515983be1)

**Comment by anjaragermai** [comment]
It gives this in WebView, I don't know if it can help but there you go and I'm not doing this for spam, you do everything when it's an English extension, but when it's French, you do nothing, for example the Asura Scans extension, With each new modification on the site you update it but when it's French you don't find anything, or sooner you don't take the trouble 
![Screenshot_20241217-105315](https://github.com/user-attachments/assets/d770af90-9567-48ee-b3a6-c7d570434e79)

**Comment by vetleledaal** [comment]
> you do everything when it's an English extension, but when it's French, you do nothing

If you want to contribute, a good way to start is by reading [the contributing guidelines](https://github.com/keiyoushi/extensions-source/blob/main/CONTRIBUTING.md). If you get stuck feel free to ask in #programming on Discord.

As you can see by the previous comments, several contributors are facing issues with loading the site in the first place. It is unreasonable to expect everyone to have a VPN at their disposal.

**Comment by quamilou** [comment]
The current URL configured for the Scan-Manga extension (https://www.scan-manga.com) returns a 500 Internal Server Error when accessed via the app and a 404 error when accessed via WebView.  
After investigation, it seems that the correct URL for the source is https://www.scan-manga.com/?mf, which points to a valid manifest.json file.  

Suggested fix: Update the extension to use the new URL (https://www.scan-manga.com/?mf) to avoid these errors. 
¯⁠\⁠_⁠(⁠ツ⁠)⁠_⁠/⁠¯

**Comment by Mylloon** [comment]
> The current URL configured for the Scan-Manga extension (https://www.scan-manga.com) returns a 500 Internal Server Error when accessed via the app and a 404 error when accessed via WebView. After investigation, it seems that the correct URL for the source is https://www.scan-manga.com/?mf, which points to a valid manifest.json file.
> 
> Suggested fix: Update the extension to use the new URL (https://www.scan-manga.com/?mf) to avoid these errors. ¯⁠\⁠_⁠(⁠ツ⁠)⁠_⁠/⁠¯

Not working on my end.

**Comment by GitShinikawa** [comment]
Hello, I'm not sure if this helps, but the extension is now visible through the Tachiyomi browser. However, nothing is showing up in the app itself. This is important because it's the French site with the largest catalog.

Thank you for your hard work!

**Comment by Passtck** [comment]
Hello, any news on this topic ? The extension still doesn't work.
Thanks everyone

**Comment by EddyOops** [comment]
Hello.
The URL has changed to https://m.scan-manga.com/
Is it possible to fix the extension?

**Comment by petiterose1** [comment]
I still can't connect personally I tried several links but it all comes back to error 404

**Comment by Norsze** [comment]
@petiterose1 this is not a forum for you to receive troubleshooting help. This is solely for development, and as you noticed the issue report is still open. Refrain from posting unnecessarily without any useful input on issues. We have a Discord for that.

**Comment by pixelroll** [comment]
now works with WebView but content still not displayed mihon

**Comment by osamu00** [comment]
Hello ,

I wanted to share my progress and ask for help regarding the Scan-Manga extension.

So far, I managed to:
- Display the popular manga list with some basic details (title, thumbnail, URL).
- Show the chapter lists correctly.
- Display the popular manga (though I haven’t fully worked on optimizing this part yet).

What I haven’t tried or implemented yet:
 
- The search bar functionality (still using the default or not implemented).
- Better handling or optimization for the popular manga tab.
- latest manga.

Where I’m stuck:

- Retrieving and decoding the pages URLs for the chapters. The pages are obfuscated using complex JavaScript with eval() and encoding techniques, making it difficult to extract the image URLs correctly.
Because of this, I get an error 

Other notes:

- The popular manga tab could use some optimization in the way  displayed.
- I am new to Kotlin and GitHub, so most of what I’ve done is thanks to ChatGPT and reading through the existing code, but I’m not confident in my ability to fix the page decoding part.
- If anyone can provide advice or help to understand how to properly decode the pages and improve the extension, I would greatly appreciate it!

By the way, I’m French and working on the French source of this extension.

Thanks for your work and support!

**Comment by Starmania** [comment]
Hey @osamu00 ! Could you create a draft PR to get an idea of what you have done ? This way we will be able to give you real suggestions. I checked the branches of your fork but I didn't see anything pushed

**Comment by osamu00** [comment]
Thanks for your reply, @Starmania .
I managed to create a branch and make a draft PR. I may not have done things correctly because this is my first time doing this. Here's the link:
https://github.com/keiyoushi/extensions-source/pull/9943
